Lib/bdb.py

```diff
@@ -26,6 +34,12 @@ def __init__(self, skip=None):
         self.frame_returning = None
 
     def canonic(self, filename):
+        """Get a filename into a canonical form.
```

Comment:
Thanks!  Three questions on this function:
1.  It seems like this helper function would be valuable in os.path, but maybe not if it hasn't been needed before.  Are there times when helper functions like this are moved?
2.  This function also caches this results.  Would there be any point in using a decorator or does that add unnecessary overhead?
3.  os.path.abspath says that it returns a normalized absolutized version of the path.  So why call both here?